### PR TITLE
fix: OCI Url and Version parse error

### LINKED ISSUE #244: In OCI mode, when the url in repositories contains a semicolon, OCI Url and Version will parse error
### Operating system

macOS 11.4

### Helmfile Version

v0.145.2

### Helm Version

v3.9.1

### Bug description

In OCI mode, when the url in repositories contains semicolon :, OCI Url and Version will parse error

### Example helmfile.yaml

```
repositories:
 - name: my-oci-repo
   url: "registry:443/helm-charts"
   oci: true

releases:
- name: nginx
  chart: my-oci-repo/nginx-ingress
  version: 0.14.0
  set:
  - name: controller.hostNetwork
    value: true
```

### Error message you've seen (if any)

```                   
Pulling registry:443/helm-charts/nginx-ingress:0.14.0
Exporting registry:443/helm-charts/nginx-ingress:0.14.0
Error: in examples/helmfile.yaml: [release "nginx": command "/usr/local/bin/helm" exited with non-zero status:

PATH:
  /usr/local/bin/helm

ARGS:
  0: helm (4 bytes)
  1: fetch (5 bytes)
  2: oci://registry (14 bytes)
  3: --version (9 bytes)
  4: 443/helm-charts/nginx-ingress (29 bytes)
  5: --destination (13 bytes)
  6: /var/folders/r1/6m48m2zj0dd2973tyc9lgxtm0000gp/T/chart3285243845 (64 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: invalid_reference: missing repository

COMBINED OUTPUT:
  Error: invalid_reference: missing repository]
in examples/helmfile.yaml: [release "nginx": command "/usr/local/bin/helm" exited with non-zero status:

PATH:
  /usr/local/bin/helm

ARGS:
  0: helm (4 bytes)
  1: fetch (5 bytes)
  2: oci://registry (14 bytes)
  3: --version (9 bytes)
  4: 443/helm-charts/nginx-ingress (29 bytes)
  5: --destination (13 bytes)
  6: /var/folders/r1/6m48m2zj0dd2973tyc9lgxtm0000gp/T/chart3285243845 (64 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: invalid_reference: missing repository

COMBINED OUTPUT:
  Error: invalid_reference: missing repository]
```

### Steps to reproduce

None

### Working Helmfile Version

None

### Relevant discussion

_No response_

### PR DESCRIPTION
**Pull request by xiaomudk**
fix #244
Signed-off-by: xiaomudk <xiaomudk@gmail.com>

Consider the following cases
```
1. registry:443/helm-charts
2. registry/helm-charts:latest
3. registry:443/helm-charts:latest
```

### CODE REVIEW

File: pkg/helmexec/exec_test.go [hunk 1]
```diff
@@ -843,3 +843,38 @@ func Test_IsVersionAtLeast(t *testing.T) {
...
+			url, tag := resolveOciChart(tt.chartPath)
+			if tt.ociChartURL != url || tt.ociChartTag != tag {
+				actual := fmt.Sprintf("ociChartURL->%s  ociChartTag->%s", url, tag)
+				expected := fmt.Sprintf("ociChartURL->%s ociChartTag->%s", tt.ociChartURL, tt.ociChartTag)
+				t.Errorf("resolveOciChart()\nactual = %v\nexpect = %v", actual, expected)
+			}
+		},
+		)
+	}
+
```
Comment (yxxhero) [inline pkg/helmexec/exec_test.go:879]: remove this space line.

File: pkg/helmexec/exec.go [hunk 2]
```diff
@@ -546,3 +542,15 @@ func (helm *execer) IsVersionAtLeast(versionStr string) bool {
 	ver := semver.MustParse(versionStr)
 	return helm.version.Equal(ver) || helm.version.GreaterThan(ver)
 }
+
+func resolveOciChart(ociChart string) (string, string) {
+	var urlTagIndex int
+	if strings.LastIndex(ociChart, ":") <= strings.LastIndex(ociChart, "/") {
```
Comment (yxxhero) [inline pkg/helmexec/exec.go:548]: can you add comments for this line?

Comment (xiaomudk) [reply pkg/helmexec/exec.go:552]: OK

File: pkg/helmexec/exec.go [hunk 3]
```diff
@@ -546,3 +542,19 @@ func (helm *execer) IsVersionAtLeast(versionStr string) bool {
 	ver := semver.MustParse(versionStr)
 	return helm.version.Equal(ver) || helm.version.GreaterThan(ver)
 }
+
+func resolveOciChart(ociChart string) (string, string) {
+	var urlTagIndex int
+	//Get the last : index
+	//e.g., registry:443/helm-charts
+	//e.g., registry/helm-charts:latest
+	//e.g., registry:443/helm-charts:latest
```
Comment (yxxhero) [inline pkg/helmexec/exec.go:551]: // Get the last : index
	// 1. registry:443/helm-charts
	// 2. registry/helm-charts:latest
	// 3. registry:443/helm-charts:latest

Comment (yxxhero) [reply pkg/helmexec/exec.go:551]: add test for all case. Thanks very much.

Comment (xiaomudk) [reply pkg/helmexec/exec.go:551]: Submitted, Thanks.

Comment (yxxhero) [reply pkg/helmexec/exec.go:551]: update comments. Thanks very much.

File: pkg/helmexec/exec_test.go (hunk 1 above)
Comment (yxxhero) [reply pkg/helmexec/exec_test.go:884]: @xiaomudk fix lint.

**Review by yxxhero — APPROVED** [review]

**Pull request merged**